Tasks/Common/webdeployment-common/kuduutility.ts

```diff
@@ -223,54 +344,70 @@ async function runCommandOnKudu(publishUrl: string, physicalPath: string, header
 }
 
-export async function runPostDeploymentScript(publishingProfile, scriptType, inlineScript, scriptPath, appOfflineFlag) {
-
-    var basicAuthToken = 'Basic ' + new Buffer(publishingProfile.userName + ':' + publishingProfile.userPWD).toString('base64');
-    var headers = {
-        'Authorization': basicAuthToken,
-        'If-Match': '*'
-    };
-    var scriptFilePath = '';
+function getPostDeploymentScript(scriptType, inlineScript, scriptPath) {
     if(scriptType === 'Inline Script') {
-        scriptFilePath = path.join(tl.getVariable('System.DefaultWorkingDirectory'),'kuduPostDeploymentScript_temp.cmd');
+        var scriptFilePath = path.join(tl.getVariable('System.DefaultWorkingDirectory'),'kuduPostDeploymentScript_local.cmd');
         tl.writeFile(scriptFilePath, inlineScript);
         tl.debug('Created temporary script file : ' + scriptFilePath);
+        return {
+            filePath: scriptFilePath,
+            isCreated: true
+        };
     }
-    else {
-        if(!tl.exist(scriptPath)) {
-            throw Error(tl.loc('ScriptFileNotFound', scriptPath));
-        }
-        var scriptExtension = path.extname(scriptPath);
-        if(scriptExtension != '.bat' && scriptExtension != '.cmd') {
-            throw Error(tl.loc('InvalidScriptFile', scriptPath));
-        }
-        scriptFilePath = scriptPath;
+    if(!tl.exist(scriptPath)) {
+        throw Error(tl.loc('ScriptFileNotFound', scriptPath));
+    }
+    var scriptExtension = path.extname(scriptPath);
+    if(scriptExtension != '.bat' && scriptExtension != '.cmd') {
+        throw Error(tl.loc('InvalidScriptFile', scriptPath));
     }
+    return {
+        filePath: scriptPath,
+        isCreated: false
+    }
+}
 
+export async function runPostDeploymentScript(publishingProfile, scriptType, inlineScript, scriptPath, appOfflineFlag) {
+    var scriptFile = getPostDeploymentScript(scriptType, inlineScript, scriptPath);
+    var uniqueID = Date.now();
+
+    var deleteLogFiles = false;
     if(appOfflineFlag) {
-        var appOfflineFilePath = path.join(tl.getVariable('system.DefaultWorkingDirectory'), 'app_offline_temp.htm');
+        var appOfflineFilePath = path.join(tl.getVariable('system.DefaultWorkingDirectory'), 'app_offline_local.htm');
         tl.writeFile(appOfflineFilePath, '<h1>App Service is offline.</h1>');
-        await uploadFiletoKudu(publishingProfile.publishUrl, '/site/wwwroot', headers, 'app_offline.htm', appOfflineFilePath);
+        await uploadFiletoKudu(publishingProfile, 'site/wwwroot', 'app_offline.htm', appOfflineFilePath);
     }
-
     try {
-        await uploadFiletoKudu(publishingProfile.publishUrl, '/site/wwwroot', headers, 'kuduPostDeploymentScript.cmd', scriptFilePath);
+        var mainCmdFilePath = path.join(tl.getVariable('system.DefaultWorkingDirectory'), 'mainFile_local.cmd');
+        tl.writeFile(mainCmdFilePath, "@echo off\ndel script_result.txt /F /Q\ncall kuduPostDeploymentScript_" + uniqueID + ".cmd" +
```

Comment:
yes, will have a static file for mainFileCmd. There is no specific reason to choose cmd over PS1. 
since, we support users to run batch commands, I planned to have the same for our script too.